### PR TITLE
manifest: update to use mbedTLS v2.16.2

### PR DESCRIPTION
**Pull request by ioannisg**
Update manifest, to point Zephyr to use
mbedTLS v2.16.2.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

---------------

https://github.com/zephyrproject-rtos/mbedtls/pull/2

### CODE REVIEW

**Comment by ioannisg** [comment]
This now needs TSC approval to get merged before 2.0 release.

**Comment by ioannisg** [comment]
We now need TSC approval to merge this to 2.0 release. @d3zd3z could you take a look and approve to have this in 2.0?

File: west.yml
Comment (d3zd3z) [inline west.yml]: Should this be a tag instead of a pull branch on github?

**Review by d3zd3z — COMMENTED** [review]
Otherwise, looks good.

File: west.yml
Comment (aescolar) [reply west.yml]: @d3zd3z Until the fixes are merged in https://github.com/zephyrproject-rtos/mbedtls this PR would just be a way of triggering CI with [the mbedtls PR](https://github.com/zephyrproject-rtos/mbedtls/pulls), not something which is meant to be merged.
But, after it is merged there, this line would need to be changed to point to the corresponding SHA, and then it could be merged here.

Comment (ioannisg) [reply west.yml]: of course

**Review by nashif — APPROVED** [review]

**Comment by ioannisg** [comment]
Pls, do not merge - we need to update the hash after the mbedtls udpate PR is merged.

**Review by carlescufi — APPROVED** [review]

File: west.yml
Comment (pfalcon) [reply west.yml]: Then I'd add `[DNM]` to a title, not just as a label...

**Review by pfalcon — APPROVED** [review]
Tested with a sample: https://github.com/zephyrproject-rtos/mbedtls/pull/2#issuecomment-525868578

**Comment by ioannisg** [comment]
> 
> 
> Tested with a sample: [zephyrproject-rtos/mbedtls#2 (comment)](https://github.com/zephyrproject-rtos/mbedtls/pull/2#issuecomment-525868578)

Thanks for your testing @pfalcon :)

**Pull request merged**
